app/code/Magento/CompareListGraphQl/etc/schema.graphqls

```diff
@@ -0,0 +1,42 @@
+# Copyright © Magento, Inc. All rights reserved.
+# See COPYING.txt for license details.
+
+type ComparableItem {
+    productId: ID! @doc(description: "Product Id")
+    name: String! @doc(description: "Product name")
+    sku: String! @doc(description: "Product SKU")
+    priceRange: PriceRange! @doc(description: "Product prices")
+    canonical_url: String @doc(description: "Product URL")
+    images: [ProductImage]! @doc(description: "Product Images")
+    values: [ProductAttribute]! @doc(description: "Product comparable attributes")
+}
+
+type ProductAttribute {
+    code: String! @doc(description:"Attribute code")
+    value: String! @doc(description:"Addibute display value")
+}
+
+type ComparableAttribute {
+    code: String! @doc(description: "Attribute code")
+    title: String! @doc(description: "Addibute display title")
+}
+
+type CompareList {
+    list_id: ID! @doc(description: "Compare list id")
+    items: [ComparableItem] @doc(description: "Comparable products")
+    attributes: [ComparableAttribute] @doc(description: "Comparable attributes, provides codes and titles for the attributes")
+}
+
+type Customer {
+    compare_list: CompareList @resolver(class: "\\Magento\\CompareListGraphQl\\Model\\Resolver\\CustomerCompareList") @doc(description: "Active customers compare list")
+}
+
+type Query {
+    compareList(id: ID!): CompareList @resolver(class: "\\Magento\\CompareListGraphQl\\Model\\Resolver\\CompareList") @doc(description: "Compare list")
+}
+
+type Mutation {
+    addItemsToCompareList(id: ID!, items: [ID!]): CompareList @resolver(class: "\\Magento\\CompareListGraphQl\\Model\\Resolver\\AddItemsToCompareList") @doc(description: "Add items to compare list")
```

Comment:
Is this id being generated on the client side? How will guest users have access to this id?

Reply:
@prabhuram93 
Unfortunately I didn't take this case into account
Could you advise please how this case should work  ?